app/src/main/java/co/nimblehq/compose/crypto/ui/screens/detail/DetailViewModel.kt

```diff
@@ -32,19 +45,67 @@ class DetailViewModel @Inject constructor(
     override val coinDetailUiModel: StateFlow<CoinDetailUiModel?>
         get() = _coinDetailUiModel
 
+    private val _coinPrices = MutableStateFlow<List<CoinPrice>>(emptyList())
+    override val coinPrices: StateFlow<List<CoinPrice>>
+        get() = _coinPrices
+
     override fun getCoinId(coinId: String) {
         getCoinDetail(coinId = coinId)
+        getCoinPrices(coinId = coinId)
     }
 
     private fun getCoinDetail(coinId: String) = execute {
         showLoading()
         getCoinDetailUseCase.execute(coinId = coinId)
             .catch { e ->
+                hideLoading()
                 _error.emit(e)
             }
             .collect { coinDetail ->
+                hideLoading()
                 _coinDetailUiModel.emit(coinDetail.toUiModel())
             }
         hideLoading()
     }
+
+    @Suppress("MagicNumber")
+    fun getCoinPrices(coinId: String, timeIntervals: TimeIntervals = TimeIntervals.ONE_DAY) =
+        execute {
+            val calendar = Calendar.getInstance()
+            val fromTimestamp = when (timeIntervals) {
+                TimeIntervals.ONE_DAY -> calendar.apply {
+                    add(Calendar.DAY_OF_YEAR, -1)
+                }
+                TimeIntervals.ONE_WEEK -> calendar.apply {
+                    add(Calendar.DAY_OF_YEAR, -7)
+                }
+                TimeIntervals.ONE_MONTH -> calendar.apply {
+                    add(Calendar.MONTH, -1)
+                }
+                TimeIntervals.ONE_YEAR -> calendar.apply {
+                    add(Calendar.YEAR, -1)
+                }
+                TimeIntervals.FIVE_YEAR -> calendar.apply {
+                    add(Calendar.YEAR, -5)
+                }
+            }
+
+            showLoading()
+            getCoinPricesUseCase.execute(
+                GetCoinPricesUseCase.Input(
+                    coinId = coinId,
+                    currency = COIN_PRICES_CURRENCY,
+                    fromTimestamp = fromTimestamp.timeInMillis.div(1000),
+                    toTimestamp = Calendar.getInstance().timeInMillis.div(1000)
+                )
+            ).catch { e ->
+                hideLoading()
+                _error.emit(e)
+            }.filterNot {
+                it.isEmpty()
+            }.collect { coinPrices ->
+                hideLoading()
+                _coinPrices.emit(coinPrices)
+            }
```

Comment:
same, `hideLoading()` at the end, please

```suggestion
            showLoading()
            getCoinPricesUseCase.execute(
                GetCoinPricesUseCase.Input(
                    coinId = coinId,
                    currency = COIN_PRICES_CURRENCY,
                    fromTimestamp = fromTimestamp.timeInMillis.div(1000),
                    toTimestamp = Calendar.getInstance().timeInMillis.div(1000)
                )
            ).catch { e ->
                _error.emit(e)
            }.filterNot {
                it.isEmpty()
            }.collect { coinPrices ->
                _coinPrices.emit(coinPrices)
            }
            hideLoading()
```